server/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/vertexcover-io/locatr"
+)
+
+var clientAndLocatrs = make(map[string]locatr.LocatrInterface)
+
+func createLocatrOptions(message incomingMessage) locatr.BaseLocatrOptions {
+	opts := locatr.BaseLocatrOptions{}
+	llmConfig := message.Settings.LlmSettings
+
+	if llmConfig.ReRankerApiKey != "" {
+		opts.ReRankClient = locatr.NewCohereClient(llmConfig.ReRankerApiKey)
+	}
+
+	opts.LogConfig = locatr.LogConfig{Level: locatr.Debug}
+
+	opts.CachePath = message.Settings.CachePath
+	opts.UseCache = message.Settings.UseCache
+
+	opts.ResultsFilePath = message.Settings.ResultsFilePath
+
+	llmClient, _ := locatr.NewLlmClient(
```

Comment:
handle error somehow.

Reply:
if not a maybe a comment on your reasoning behind why it need not be handled here.